modules/reference/pages/properties/object-storage-properties.adoc

```diff
@@ -0,0 +1,1494 @@
+
+Optional API endpoint.
+
+- AWS: when left blank, it's automatically generated using <<cloud_storage_region,region>> property, and <<cloud_storage_bucket,bucket>> property. Otherwise uses the value assigned.
+- GCP: use `storage.googleapis.com`.
+
+*Requires restart:* No
+
+*Optional:* No
+
+*Visibility:* `user`
+
+*Type:* string
+
+*Default:* `null`
+
+---
+
+=== cloud_storage_api_endpoint_port
+
+TLS port override.
+
+*Requires restart:* No
+
+*Optional:* Yes
+
+*Visibility:* `user`
+
+*Type:* integer
+
+*Accepted values:* [`-32768`, `32767`]
+
+*Default:* `443`
+
+---
+
+=== cloud_storage_bucket
+
+AWS or GCS bucket that should be used to store data.
+
+*Requires restart:* No
+
+*Optional:* No
+
+*Visibility:* `user`
+
+*Type:* string
+
+*Default:* `null`
+
+---
+
+
+=== cloud_storage_cache_size
+
+Max size of object storage cache.
+
+*Units*: bytes.
+
+*Requires restart:* No
+
+*Optional:* Yes
+
+*Visibility:* `user`
+
+*Type:* integer
+
+*Accepted values:* [`0`, `18446744073709551615`]
+
+*Default:* `0`
+
+
+---
+
+=== cloud_storage_cluster_metadata_upload_interval_ms
+
+Time interval to wait between cluster metadata uploads.
+
+*Units*: milliseconds.
+
+*Requires restart:* No
+
+*Optional:* Yes
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `3600000` (1 hour)
+
+---
+
+=== cloud_storage_credentials_source
+
+The source of credentials used to authenticate to object storage services.
+Required for AWS and GCS authentication with IAM roles.
+
+To authenticate using access keys, see <<cloud_storage_access_key>>.
+
+*Valid values*: `config_file`, `aws_instance_metadata`, `sts, gcp_instance_metadata`, `azure_vm_instance_metadata`, `azure_aks_oidc_federation`
+
+*Requires restart:* Yes
+
+*Optional:* Yes
+
+*Visibility:* `user`
+
+*Default:* `config_file`
+
+---
+
+=== cloud_storage_disable_tls
+
+Disable TLS for all S3 or GCS connections.
+
+*Type*: boolean
+
+*Default*: false
+
+*Restart required*: yes
+
+---
+
+=== cloud_storage_enabled
+
+Enable object storage. Must be set to `true` to use xref:manage:tiered-storage.adoc[Tiered Storage] or Remote Read Replicas.
+
+*Type*: boolean.
+
+*Requires restart:* No
+
+*Optional:* Yes
+
+*Visibility:* `user`
+
+*Type:* boolean
+
+*Default:* `false`
+
+---
+
+=== cloud_storage_max_connections
+
+Maximum simultaneous S3 connections per shard, applicable to upload and download activities.
+
+*Units*: number of simultaneous connections.
+
+*Requires restart:* No
+
+*Optional:* Yes
+
+*Visibility:* `user`
+
+*Type:* integer
+
+*Accepted values:* [`-32768`, `32767`]
+
+*Default:* `20`
+
+---
+
+=== cloud_storage_recovery_topic_validation_mode
+
+Validation performed before recovering a topic from cloud storage. In case of failure, the reason for the failure appears as `ERROR` lines in the Redpanda application log. For each topic, this reports errors for all partitions, but for each partition, only the first error is reported.
```

Comment:
```suggestion
Validation performed before recovering a topic from object storage. In case of failure, the reason for the failure appears as `ERROR` lines in the Redpanda application log. For each topic, this reports errors for all partitions, but for each partition, only the first error is reported.
```